### ISSUE TITLE
Crash with ShowMeHills when using as compile-dependency

### ISSUE THREAD

**Issue by dalb8** [mar-v-in/MapsAPI#5]
I built https://github.com/ShowMeHills/ShowMeHills.git from source with MapsAPI instead of Google APIs. When I launch the maps activity I get a crash. 
Fdroid recipe is like this:

```
Build:0.30,30
    commit=v0.30
    subdir=ShowMe/ShowMeHills
    init=sed -i '/uses-library/d' AndroidManifest.xml && \
      sed -i 's@\(target=\).*@\1android-10@' ../ShowMeX/project.properties && \
    sed -i 's@\(target=\).*@\1android-10@' project.properties
    srclibs=2:MapsAPI@323b519,NoAnalytics@d38ae535b543
    rm=ShowMe/ShowMeX/libs/libGoogleAnalyticsV2.jar,ShowMe/ShowMeHills/bin
    prebuild=mkdir src && \
        echo "android.library.reference.1=$$NoAnalytics$$" >> ../ShowMeX/project.properties && \
        echo "android.library.reference.2=$$MapsAPI$$" >> ../ShowMeX/project.properties
```

or as diff:

```
diff --git a/ShowMe/ShowMeHills/AndroidManifest.xml b/ShowMe/ShowMeHills/AndroidManifest.xml
index 19dee15..c23292f 100644
--- a/ShowMe/ShowMeHills/AndroidManifest.xml
+++ b/ShowMe/ShowMeHills/AndroidManifest.xml
@@ -17,7 +17,6 @@
    <uses-feature android:name="android.hardware.camera" /> 

     <application android:icon="@drawable/icon" android:label="@string/app_name" android:name="ShowMeApplication">
-       <uses-library android:name="com.google.android.maps" />
         <activity android:name=".ShowMeHillsActivity"
                   android:label="@string/app_name"
                   android:theme="@android:style/Theme.NoTitleBar" 
diff --git a/ShowMe/ShowMeHills/project.properties b/ShowMe/ShowMeHills/project.properties
index db22827..659e515 100644
--- a/ShowMe/ShowMeHills/project.properties
+++ b/ShowMe/ShowMeHills/project.properties
@@ -11,5 +11,6 @@
 #proguard.config=${sdk.dir}/tools/proguard/proguard-android.txt:proguard-project.txt

 # Project target.
-target=Google Inc.:Google APIs:10
+target=android-10
 android.library.reference.1=../ShowMeX
+android.library.reference.2=../../../srclib/MapsAPI/com.google.android.maps
diff --git a/ShowMe/ShowMeX/libs/libGoogleAnalyticsV2.jar b/ShowMe/ShowMeX/libs/libGoogleAnalyticsV2.jar
deleted file mode 100644
index 46d0e44..0000000
Binary files a/ShowMe/ShowMeX/libs/libGoogleAnalyticsV2.jar and /dev/null differ
diff --git a/ShowMe/ShowMeX/project.properties b/ShowMe/ShowMeX/project.properties
index 045e57a..c18371d 100644
--- a/ShowMe/ShowMeX/project.properties
+++ b/ShowMe/ShowMeX/project.properties
@@ -11,5 +11,7 @@
 # Project target.
-target=Google Inc.:Google APIs:10
+target=android-10
 android.library=true
+android.library.reference.1=../../../srclib/NoAnalytics/NoAnalytics
+android.library.reference.2=../../../srclib/MapsAPI/com.google.android.maps
```

```
/formResponse?formkey=dGZ5b0dtQl9WRDNBMF9DXzJ3UGNZT2c6MQ&amp;ifq
W/ResponseProcessCookies( 6028): Invalid cookie header: "Set-Cookie: NID=67=Gi_UKmR1yi85I7evpANORnoET3LiUVRJvrIssCbb22hxV9zbFrRUKBZhn-tvjhK24D-9Lawljyvh7b3IK5_luV4UUF0nrbSPChnPFQcLAJJZkzKvbd6R_ywFN_3Ycoo7;Domain=.google.com;Path=/;Expires=Sat, 13-Sep-2014 10:05:51 GMT;HttpOnly". Unable to parse expires attribute: Sat
D/ACRA    ( 6028): #checkAndSendReports - finish
E/AndroidRuntime( 6028): FATAL EXCEPTION: main
E/AndroidRuntime( 6028): java.lang.NullPointerException
E/AndroidRuntime( 6028):    at com.showmehills.MapOverlayCompassItem.draw(MapOverlayCompassItem.java:59)
E/AndroidRuntime( 6028):    at com.google.android.maps.OverlayManager.onDraw(OverlayManager.java:124)
E/AndroidRuntime( 6028):    at com.google.android.maps.MapView.dispatchDraw(MapView.java:848)
E/AndroidRuntime( 6028):    at android.view.View.draw(View.java:13601)
E/AndroidRuntime( 6028):    at android.view.ViewGroup.drawChild(ViewGroup.java:2928)
E/AndroidRuntime( 6028):    at android.view.ViewGroup.dispatchDraw(ViewGroup.java:2797)
E/AndroidRuntime( 6028):    at android.view.View.draw(View.java:13601)
E/AndroidRuntime( 6028):    at android.view.ViewGroup.drawChild(ViewGroup.java:2928)
E/AndroidRuntime( 6028):    at android.view.ViewGroup.dispatchDraw(ViewGroup.java:2797)
E/AndroidRuntime( 6028):    at android.view.View.draw(View.java:13601)
E/AndroidRuntime( 6028):    at android.view.ViewGroup.drawChild(ViewGroup.java:2928)
E/AndroidRuntime( 6028):    at android.view.ViewGroup.dispatchDraw(ViewGroup.java:2797)
E/AndroidRuntime( 6028):    at android.view.View.draw(View.java:13722)
E/AndroidRuntime( 6028):    at android.widget.FrameLayout.draw(FrameLayout.java:467)
E/AndroidRuntime( 6028):    at com.android.internal.policy.impl.PhoneWindow$DecorView.draw(PhoneWindow.java:2473)
E/AndroidRuntime( 6028):    at android.view.ViewRootImpl.drawSoftware(ViewRootImpl.java:2281)
E/AndroidRuntime( 6028):    at android.view.ViewRootImpl.draw(ViewRootImpl.java:2177)
E/AndroidRuntime( 6028):    at android.view.ViewRootImpl.performDraw(ViewRootImpl.java:2045)
E/AndroidRuntime( 6028):    at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:1854)
E/AndroidRuntime( 6028):    at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:989)
E/AndroidRuntime( 6028):    at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:4351)
E/AndroidRuntime( 6028):    at android.view.Choreographer$CallbackRecord.run(Choreographer.java:749)
E/AndroidRuntime( 6028):    at android.view.Choreographer.doCallbacks(Choreographer.java:562)
E/AndroidRuntime( 6028):    at android.view.Choreographer.doFrame(Choreographer.java:532)
E/AndroidRuntime( 6028):    at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:735)
E/AndroidRuntime( 6028):    at android.os.Handler.handleCallback(Handler.java:725)
E/AndroidRuntime( 6028):    at android.os.Handler.dispatchMessage(Handler.java:92)
E/AndroidRuntime( 6028):    at android.os.Looper.loop(Looper.java:137)
E/AndroidRuntime( 6028):    at android.app.ActivityThread.main(ActivityThread.java:5227)
E/AndroidRuntime( 6028):    at java.lang.reflect.Method.invokeNative(Native Method)
E/AndroidRuntime( 6028):    at java.lang.reflect.Method.invoke(Method.java:511)
E/AndroidRuntime( 6028):    at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:795)
E/AndroidRuntime( 6028):    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:562)
E/AndroidRuntime( 6028):    at dalvik.system.NativeStart.main(Native Method)
W/ActivityManager(  377):   Force finishing activity com.showmehills/.MapOverlay
D/dalvikvm(  377): GC_FOR_ALLOC freed 400K, 18% free 9284K/11264K, paused 46ms, total 47ms
W/ActivityManager(  377):   Force finishing activity com.showmehills/.ShowMeHillsActivity
W/ActivityManager(  377): Activity pause timeout for ActivityRecord{411a0850 u0 com.showmehills/.MapOverlay}
D/dalvikvm(  607): GC_EXPLICIT freed 280K, 17% free 6428K/7700K, paused 2ms+12ms, total 47ms
E/Trebuchet.AppsCustomizePagedView(  607): Widget ComponentInfo{org.videolan.vlc/org.videolan.vlc.widget.VLCAppWidgetProvider} can not fit on this device (441, 60)
D/LiveWallpaper(  563): onVisibilityChanged:true
D/LiveWallpaper(  563): check SD:/storage/sdcard/BeautyClock/pic/clockm/0015.jpg
D/dalvikvm(  563): GC_FOR_ALLOC freed 937K, 62% free 2970K/7700K, paused 17ms, total 17ms
I/dalvikvm-heap(  563): Grow heap (frag case) to 4.540MB for 957616-byte allocation
D/dalvikvm(  563): GC_CONCURRENT freed <1K, 50% free 3904K/7700K, paused 3ms+2ms, total 16ms
D/dalvikvm(  563): GC_CONCURRENT freed 538K, 50% free 3873K/7700K, paused 4ms+1ms, total 22ms
E/Trebuchet.Workspace(  607): The screen must be >= 0 and < 1 (was 2); skipping child
E/Trebuchet.Workspace(  607): The screen must be >= 0 and < 1 (was 3); skipping child
E/Trebuchet.Workspace(  607): The screen must be >= 0 and < 1 (was 3); skipping child
```

**Comment by mar-v-in** [comment]
Thanks for pointing out, should be fixed by 97d1ac514f74a7c7097bbe375889db6423ceda48.

**Comment by dalb8** [comment]
Working fine, thanks.
